db/migrate/20170821035759_add_original_to_project.rb

```diff
@@ -0,0 +1,8 @@
+class AddOriginalToProject < ActiveRecord::Migration
+  def change
+    add_column :projects, :original_id, :integer, polymorphic: true, foreign_key: { references: :projects}
+    # add_reference :projects, :original
+    # add_foreign_key :projects, :original, column: :original_id
+    # add_index :projects, [:id, :original_id]
```

Comment:
I meant to remove the polymorphic, thanks for catching.
The foreign key is an index, so we don't need to add one. I checked psql, and it is there.